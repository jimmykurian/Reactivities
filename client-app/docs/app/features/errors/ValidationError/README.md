[**Client App Documentation v0.0.0**](../../../../README.md) â€¢ **Docs**

***

[Client App Documentation v0.0.0](../../../../README.md) / app/features/errors/ValidationError

# app/features/errors/ValidationError

## Index

### Interfaces

- [Props](interfaces/Props.md)

### Functions

- [default](functions/default.md)
