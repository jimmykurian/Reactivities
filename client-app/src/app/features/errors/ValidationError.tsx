/**
 * @file ValidationError.tsx
 * @author Jimmy Kurian
 * @fileoverview This file contains the ValidationError component, which displays a list of validation errors in a user-friendly format.
 */

import { Message } from 'semantic-ui-react';

/**
 * @interface Props
 * @description The props for the ValidationError component.
 * @property {string[]} errors - An array of error messages to be displayed. Each string represents an individual error message that will be shown to the user.
 */
export interface Props {
  errors: string[];
}

/**
 * ValidationError component displays a list of error messages in a red error message box.
 * This component is primarily used for displaying form validation errors.
 *
 * @component
 * @param {Props} props - The props for the ValidationError component.
 * @returns {JSX.Element} The rendered ValidationError component.
 *
 * @example
 * ```tsx
 * <ValidationError errors={['Email is required', 'Password must be at least 6 characters']} />
 * ```
 */
export default function ValidationError({ errors }: Props): JSX.Element {
  return (
    <Message error>
      {errors && (
        <Message.List>
          {errors.map((err: string, i) => (
            <Message.Item key={i}>{err}</Message.Item>
          ))}
        </Message.List>
      )}
    </Message>
  );
}
