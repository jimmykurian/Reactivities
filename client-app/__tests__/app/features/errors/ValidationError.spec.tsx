import { render, screen } from '@testing-library/react';
import ValidationError from '../../../../src/app/features/errors/ValidationError';
import { faker } from '@faker-js/faker';

describe('ValidationError Component', () => {
  it('renders without errors if no errors are provided', () => {
    // Arrange
    const errors: string[] = [];

    // Act
    render(<ValidationError errors={errors} />);

    // Assert
    const list = screen.queryByRole('list');
    expect(list).toBeInTheDocument();
    expect(screen.queryAllByRole('listitem')).toHaveLength(0);
  });

  it('renders a list of error messages', () => {
    // Arrange
    const errors = [
      faker.lorem.sentence(),
      faker.lorem.sentence(),
      faker.lorem.sentence(),
    ];

    // Act
    render(<ValidationError errors={errors} />);

    // Assert
    const listItems = screen.getAllByRole('listitem');
    expect(listItems).toHaveLength(errors.length);
    listItems.forEach((item, index) => {
      expect(item).toHaveTextContent(errors[index]);
    });
  });

  it('renders Semantic UI Message component with error class', () => {
    // Arrange
    const errors = [faker.lorem.sentence()];

    // Act
    render(<ValidationError errors={errors} />);

    // Assert
    const messageComponent = screen.getByText(errors[0]).closest('.ui.message');
    expect(messageComponent).toHaveClass('error');
  });

  it('matches the snapshot', () => {
    // Arrange
    const errors = [
      'Email is required',
      'Password must be at least 6 characters',
    ];

    // Act
    const { asFragment } = render(<ValidationError errors={errors} />);

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
});
