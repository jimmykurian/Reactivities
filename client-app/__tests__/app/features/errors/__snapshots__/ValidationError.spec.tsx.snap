// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationError Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui error message"
  >
    <ul
      class="list"
    >
      <li
        class="content"
      >
        Email is required
      </li>
      <li
        class="content"
      >
        Password must be at least 6 characters
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
